### PR TITLE
Fixes generated peform InputOutput param sockets

### PR DESCRIPTION
**Pull request by JLLeitschuh**
Mats set as InputOutput param types now use `assignTo` instead of `setTo`

### CODE REVIEW

**Comment by ThomasJClark** [comment]
What does this fix?

**Comment by JLLeitschuh** [comment]
Set to doesn't work. Set to is designed to be used with Scalars. As an example: `mat.setTo(new Mat(new Scalar(3)));`

**Comment by ThomasJClark** [comment]
Ok, but what's actually fixed now?  I don't think I understand what was wrong

**Comment by JLLeitschuh** [comment]
This is why rectangle wasn't working.

**Comment by JLLeitschuh** [comment]
I just had to fix a comment that was wrong. This is read to merge.

File: buildSrc/src/main/java/edu/wpi/gripgenerator/settings/DefinedParamType.java [hunk 1]
```diff
@@ -66,8 +67,8 @@ public DefinedParamType(String type, Parameter param) {
     }
 
     public DefinedParamType(String type, DefinedParamState state) {
-        this.type = type;
-        this.state = state;
+        this(type);
+        this.state = Optional.ofNullable(state);
```
Comment (ThomasJClark) [inline buildSrc/src/main/java/edu/wpi/gripgenerator/settings/DefinedParamType.java]: Why is the state optional?  It seems like something like that would be required.  What happens to a socket if it doesn't have a state?  (Also, why is it called "state" and not "direction"?)

Comment (JLLeitschuh) [inline buildSrc/src/main/java/edu/wpi/gripgenerator/settings/DefinedParamType.java]: I goofed up when I amended my commit. That's not part of the PR now.

**Comment by JLLeitschuh** [comment]
Okay, now it's ready. :p
